src/backend/distributed/commands/multi_copy.c

```diff
@@ -3105,32 +3222,46 @@ GetShardState(uint64 shardId, HTAB *shardStateHash,
 static void
 InitializeCopyShardState(CopyShardState *shardState,
 						 HTAB *connectionStateHash, uint64 shardId,
-						 bool stopOnFailure)
+						 bool stopOnFailure, bool shouldUseLocalCopy, MemoryContext
+						 context)
 {
 	ListCell *placementCell = NULL;
 	int failedPlacementCount = 0;
 
+	MemoryContext oldContext = MemoryContextSwitchTo(context);
+
+	MemoryContextSwitchTo(oldContext);
+
 	MemoryContext localContext =
 		AllocSetContextCreateExtended(CurrentMemoryContext,
 									  "InitializeCopyShardState",
 									  ALLOCSET_DEFAULT_MINSIZE,
 									  ALLOCSET_DEFAULT_INITSIZE,
 									  ALLOCSET_DEFAULT_MAXSIZE);
 
+
 	/* release active placement list at the end of this function */
-	MemoryContext oldContext = MemoryContextSwitchTo(localContext);
+	oldContext = MemoryContextSwitchTo(localContext);
 
 	List *activePlacementList = ActiveShardPlacementList(shardId);
 
 	MemoryContextSwitchTo(oldContext);
 
 	shardState->shardId = shardId;
 	shardState->placementStateList = NIL;
+	shardState->localCopyBuffer = makeStringInfo();
```

Comment:
In my local I get some warnings related to memory context:
![image](https://user-images.githubusercontent.com/17179875/75777650-61ad8d80-5d67-11ea-8940-f5415a309e86.png)

The localCopyBuffer is allocated in the current memory, which is PortalContext. We should probably fix that even though it doesnt happen in circleci.

Reply:
this seems to be resolved